### PR TITLE
Fix WhoReacted

### PR DESCRIPTION
**Pull request by TheKodeToad**
"Fix"

(the code is probably somehow worse)

### CODE REVIEW

File: src/plugins/whoReacted/index.tsx [hunk 1]
```diff
@@ -98,8 +95,14 @@ export default definePlugin({
     patches: [{
         find: ",reactionRef:",
         replacement: {
-            match: /(?<=(\i)=(\i)\.hideCount,)(.+?reactionCount.+?\}\))/,
-            replace: (_, hideCount, props, rest) => `whoReactedProps=${props},${rest},${hideCount}?null:$self.renderUsers(whoReactedProps)`
+            match: /(?<=(?<hideCount>\i)=(?<props>\i)\.hideCount,)(?<rest>.+?reactionCount.+?\}\))/,
+            replace: "whoReactedProps=$<props>,$<rest>,$<hideCount>?null:$self.renderUsers(whoReactedProps)"
```
[the remark this answers is not in the document]
Comment (Vendicated) [reply src/plugins/whoReacted/index.tsx:99]: i find named groups very unreadable, i think the previous code was better

File: src/plugins/whoReacted/index.tsx [hunk 2]
```diff
@@ -98,8 +95,14 @@ export default definePlugin({
...
+        }
+    }, {
+        find: ".displayName=\"MessageReactionsStore\";",
```
Comment (Vendicated) [inline src/plugins/whoReacted/index.tsx:102]: avoid escaping
```suggestion
        find: '.displayName="MessageReactionsStore";',
```

File: src/plugins/whoReacted/index.tsx [hunk 3]
```diff
@@ -29,9 +29,8 @@ import { ReactionEmoji, User } from "discord-types/general";
 const UserSummaryItem = LazyComponent(() => findByCode("defaultRenderUser", "showDefaultAvatarsForNullUsers"));
 const AvatarStyles = findByPropsLazy("moreUsers", "emptyUser", "avatarContainer", "clickableAvatar");
 
-const ReactionStore = findByPropsLazy("getReactions");
-
 const queue = new Queue();
+let reactions: Record<string, any>;
```
Comment (Vendicated) [inline src/plugins/whoReacted/index.tsx:33]: properly type this to more easily catch errors

File: src/plugins/whoReacted/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheKodeToad) [reply src/plugins/whoReacted/index.tsx:99]: Oh, I thought you were using this now - I'm pretty sure I'd seen another plugin migrate to it

File: src/plugins/whoReacted/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# WhoReacted
+
+Next to each reaction, display each user's avatar. Each avatar can be clicked and will open the profile.
+
+![image](https://github.com/Vendicated/Vencord/assets/57493648/97fec9e8-396f-4f5e-916e-1ec21445113d)
```
Comment (Vendicated) [inline src/plugins/whoReacted/README.md:5]: either add a meaningful alt text or leave it empty
```suggestion
![](https://github.com/Vendicated/Vencord/assets/57493648/97fec9e8-396f-4f5e-916e-1ec21445113d)
```

Comment (TheKodeToad) [reply src/plugins/whoReacted/README.md:5]: This is used in other READMEs and is the default.

**Pull request merged**